### ISSUE TITLE
pydoop issues on Amazon EMR

### ISSUE THREAD

**Issue by mheppner** [crs4/pydoop#167]
On a node running EMR 4.1.0 (Hadoop 2.6.0), creating an instance of pydoop.hdfs.hdfs() encounters an error:

```
>>> from pydoop import hdfs
>>> hdfs.hdfs()
loadFileSystems error:
(unable to get stack trace for java.lang.NoClassDefFoundError exception: ExceptionUtils::getStackTrace error.)
hdfsBuilderConnect(forceNewInstance=0, nn=default, port=0, kerbTicketCachePath=(NULL), userName=(NULL)) error:
(unable to get stack trace for java.lang.NoClassDefFoundError exception: ExceptionUtils::getStackTrace error.)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/hadoop/sigma-api/lib/python2.7/site-packages/pydoop-1.1.0-py2.7.egg/pydoop/hdfs/fs.py", line 150, in __init__
    h, p, u, fs = _get_connection_info(host, port, user)
  File "/home/hadoop/sigma-api/lib/python2.7/site-packages/pydoop-1.1.0-py2.7.egg/pydoop/hdfs/fs.py", line 64, in _get_connection_info
    fs = core_hdfs_fs(host, port, user)
  File "/home/hadoop/sigma-api/lib/python2.7/site-packages/pydoop-1.1.0-py2.7.egg/pydoop/hdfs/core/__init__.py", line 57, in core_hdfs_fs
    return _CORE_MODULE.CoreHdfsFs(host, port, user)
RuntimeError: (255, 'Unknown error 255')
```

Building pydoop with the JPype connector gives a better error message:

```
>>> from pydoop import hdfs
>>> hdfs.hdfs()
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/hadoop/sigma-api/lib/python2.7/site-packages/pydoop-1.1.0-py2.7.egg/pydoop/hdfs/fs.py", line 150, in __init__
    h, p, u, fs = _get_connection_info(host, port, user)
  File "/home/hadoop/sigma-api/lib/python2.7/site-packages/pydoop-1.1.0-py2.7.egg/pydoop/hdfs/fs.py", line 64, in _get_connection_info
    fs = core_hdfs_fs(host, port, user)
  File "/home/hadoop/sigma-api/lib/python2.7/site-packages/pydoop-1.1.0-py2.7.egg/pydoop/hdfs/core/__init__.py", line 57, in core_hdfs_fs
    return _CORE_MODULE.CoreHdfsFs(host, port, user)
  File "/home/hadoop/sigma-api/lib/python2.7/site-packages/pydoop-1.1.0-py2.7.egg/pydoop/hdfs/core/bridged/hadoop.py", line 76, in __init__
    JavaClassName.Configuration, True
  File "/home/hadoop/sigma-api/lib/python2.7/site-packages/pydoop-1.1.0-py2.7.egg/pydoop/hdfs/core/bridged/common.py", line 30, in wrap_class_instance
    return FACTORY.get_wrapper(fully_qualified_class_name)(*args)
  File "/home/hadoop/sigma-api/lib/python2.7/site-packages/pydoop-1.1.0-py2.7.egg/pydoop/utils/bridge/factory.py", line 59, in get_wrapper
    return self._loader.load_class(fully_qualified_class)
  File "/home/hadoop/sigma-api/lib/python2.7/site-packages/pydoop-1.1.0-py2.7.egg/pydoop/utils/bridge/jpype_loader.py", line 42, in load_class
    return jpype.JClass(self.process_class_name(fully_qualified_class))
  File "/home/hadoop/sigma-api/lib/python2.7/site-packages/jpype/_jclass.py", line 55, in JClass
    raise _RUNTIMEEXCEPTION.PYEXC("Class %s not found" % name)
jpype._jexception.RuntimeExceptionPyRaisable: java.lang.RuntimeException: Class org.apache.hadoop.conf.Configuration not found
```

I think the classpath isn't set correctly. I set the output from `hadoop classpath` as the CLASSPATH environment variable, rebuilt, and reinstalled. In the pydoop source, I noticed it's [looking for a lowercase](https://github.com/crs4/pydoop/blob/1.1.0/pydoop/hdfs/core/bridged/__init__.py#L37) env variable, which I also exported, rebuilt, and reinstalled.

Also related, in order to build pydoop, I had to modify [setup.py](https://github.com/crs4/pydoop/blob/1.1.0/setup.py#L297) to add in the classpath returned from `hadoop classpath`. Without doing this, setup.py failed to build since it couldn't find any of the Hadoop classes.

Relevant:

```
>>> pydoop.hadoop_classpath()
'/usr/lib/hadoop/lib/native:/usr/lib/hadoop/etc/hadoop'
```

```
[hadoop@localhost ~]$ hadoop classpath
/usr/lib/hadoop/etc/hadoop:/usr/lib/hadoop/lib/*:/usr/lib/hadoop/.//*:/usr/lib/hadoop-hdfs/./:/usr/lib/hadoop-hdfs/lib/*:/usr/lib/hadoop-hdfs/.//*:/usr/lib/hadoop-yarn/lib/*:/usr/lib/hadoop-yarn/.//*:/usr/lib/hadoop-mapreduce/lib/*:/usr/lib/hadoop-mapreduce/.//*::/usr/lib/hadoop-lzo/lib/*:/usr/share/aws/emr/emrfs/conf:/usr/share/aws/emr/emrfs/lib/*:/usr/share/aws/emr/emrfs/auxlib/*:/usr/share/aws/emr/ddb/lib/emr-ddb-hadoop.jar:/usr/share/aws/emr/goodies/lib/emr-hadoop-goodies.jar:/usr/share/aws/emr/kinesis/lib/emr-kinesis-hadoop.jar:/usr/share/aws/emr/cloudwatch-sink/lib/*
[hadoop@localhost ~]$ echo $HADOOP_HOME && echo $HADOOP_CONF_DIR && echo $HADOOP_VERSION
/usr/lib/hadoop
/usr/lib/hadoop/etc/hadoop
2.6.0
```

**Comment by jerrypaytm** [comment]
This is still a problem installing pydoop in EMR. Any update on this?

**Comment by simleo** [comment]
The classpath was being set incorrectly both at compile time and at runtime (hence the hdfs failures). This has been addressed in #293.
